internal/ctrlutils/ingress-status.go

```diff
@@ -32,71 +34,100 @@ const (
 	statusUpdateWaitTick = time.Second
 )
 
-// PullConfigUpdate is a dedicated function that process ingress/customer resource status update after configuration is updated within kong.
-func PullConfigUpdate(
-	ctx context.Context,
-	kongConfig sendconfig.Kong,
-	log logr.Logger,
-	kubeConfig *rest.Config,
-	publishService string,
-	publishAddresses []string,
-) {
-	var hostname string
-	var ips []string
-	for {
-		var err error
-		ips, hostname, err = RunningAddresses(ctx, kubeConfig, publishService, publishAddresses)
-		if err != nil {
-			// in the case that a LoadBalancer type service is being used for the Kong Proxy
-			// we must wait until the IP address if fully provisioned before we will be able
-			// to use the reference IPs/Hosts from that LoadBalancer to update resource status addresses.
-			if err.Error() == proxyLBNotReadyMessage {
-				log.V(util.InfoLevel).Info("LoadBalancer type Service for the Kong proxy is not yet provisioned, waiting...", "service", publishService)
-				time.Sleep(time.Second)
-				continue
-			}
+var (
+	errLBNotReady = errors.New("LoadBalancer service has no IPs assigned")
+)
 
-			log.Error(err, "failed to determine kong proxy external ips/hostnames.")
-			return
-		}
-		break
-	}
+type statusConfig struct {
+	ready             bool
+	cli               *clientset.Clientset
```

Comment:
`cli` implies "command-line interface" to me. `client`?

also applies to `kiccli`. `kicClient`?

Reply:
In this section of the code, that was the convention used elsewhere (inconsistently, but `cli` was more common than `client`):

```
08:50:15-0800 esenin $ git describe --tags
v2.0.6
08:50:18-0800 esenin $ grep -iPe "cli( |\pP)" internal/ctrlutils/ingress-status.go
	cli, err := clientset.NewForConfig(kubeConfig)
	versionInfo, err := cli.ServerVersion()
	kiccli, err := kicclientset.NewForConfig(kubeConfig)
				if err := UpdateStatuses(ctx, &updateDone, log, cli, kiccli, &wg, ips, hostname, kubeConfig, kubernetesVersion); err != nil {
	cli *clientset.Clientset,
	kiccli *kicclientset.Clientset,
			if err := UpdateTCPIngress(ctx, log, svc, kiccli, ips); err != nil {
			if err := UpdateUDPIngress(ctx, log, svc, kiccli, ips); err != nil {
					if err := UpdateIngress(ctx, log, route, cli, ips); err != nil {
					if err := UpdateIngressLegacy(ctx, log, route, cli, ips); err != nil {
	cli *clientset.Clientset,
	ingCli := cli.NetworkingV1().Ingresses(namespace)
		curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
		_, err = ingCli.UpdateStatus(ctx, curIng, metav1.UpdateOptions{})
	cli *clientset.Clientset,
	ingCli := cli.NetworkingV1beta1().Ingresses(namespace)
		curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
		_, err = ingCli.UpdateStatus(ctx, curIng, metav1.UpdateOptions{})
func UpdateUDPIngress(ctx context.Context, log logr.Logger, svc file.FService, kiccli *kicclientset.Clientset,
		ingCli := kiccli.ConfigurationV1beta1().UDPIngresses(namespace)
			curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
			_, err = ingCli.UpdateStatus(ctx, curIng, metav1.UpdateOptions{})
func UpdateTCPIngress(ctx context.Context, log logr.Logger, svc file.FService, kiccli *kicclientset.Clientset,
		ingCli := kiccli.ConfigurationV1beta1().TCPIngresses(namespace)
			curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
			_, err = ingCli.UpdateStatus(ctx, curIng, metav1.UpdateOptions{})
		knativeCli, err := knativeversioned.NewForConfig(kubeCfg)
		ingClient := knativeCli.NetworkingV1alpha1().Ingresses(namespace)
```

Unless something is egregiously incomprehensible, I'm usually not one to try and update variable name conventions unless that's exactly what I set out to do, but whatever, e5fcee77